spicy/include/rt/parsed-unit.h

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2020 by the Zeek Project. See LICENSE for details.
+
+#pragma once
+
+#include <hilti/rt/type-info.h>
+#include <hilti/rt/types/reference.h>
+
+namespace spicy::rt {
+
+/**
+ * Type-erased wrapper around parsed unit instances.
+ *
+ * Initially, this will be unbound, i.e., not refer to any particular
+ * instance. `init()` then binds it to an instance, which will let
+ * `ParsedUnit` hold a strong reference to it.
```

Comment:
Not sure I understand the motivation for this. Couldn't we just as well provide a helper function which a `ParsedUnit` which could then use RAII to manage its data? This would lead to different visitor impls though.

Reply:
The main reason here is the API that we have for calling into the Spicy-generated parsing. We can't easily pass complex objects back from there, that's why we pass a reference parameter of type `ParsedUnit` for the function to then initialize. However, the instance needs to survive the initial call into parsing so that resuming will continue to have access to it. And internally parsed instances are manager through a `ValueReference`. Hence the `StrongReference` to keep it around; a strong reference can bind to a value reference.

Does that help explain it? If you would still do it differently, can you give a quick example of how that would look like?